src/components/KanbanComponent/RepoFormComponent/RepoFormComponent.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import { Button, Grid, TextField, makeStyles } from '@material-ui/core';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    '& .MuiFormControl-root': {
+      margin: theme.spacing(3),
+    },
+  },
+}));
+
+export const RepoFormComponent = ({ username }: { username: string | undefined }) => {
+  const initialFormValues = {
+    ghUsername: username,
+    repositoryName: '',
+  };
+  const [formValues, setFormValues] = useState(initialFormValues);
+  const [formVisible, setFormVisible] = useState(false);
+
+  const requestOptions = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      user_id: formValues.ghUsername,
+      repository: formValues.repositoryName,
+    }),
+  };
+
+  const classes = useStyles();
+
+  const handleClick = () => {
+    console.log('handle click');
+    setFormVisible(formVisible ? false : true);
```

Comment:
Do this instead. [This](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state) article explains the difference. If you still want to use the set functions without the updater function I'd suggest this: `setFormVisible(!formVisible);`
```suggestion
    setFormVisible(prevFormVisible => !prevFormVisible);
```